docs/content/documentation/systems/AuxKernels/tensor_mechanics/RankFourAux.md

```diff
@@ -1,8 +1,40 @@
-<!-- MOOSE Documentation Stub: Remove this when content is added. -->
-
-# RankFourAux
+# Rank Four Aux
 !syntax description /AuxKernels/RankFourAux
 
+## Description
+The AuxKernel `RankFourAux` is used to save single components of Rank-4 tensors into an AuxVariable for visualization and/or post-processing purposes, similar to the functionality provided by [RankTwoAux](/RankTwoAux.md) for Rank-2 tensors
+`RankFourAux` is commonly used to output components of the elasticity (or stiffness) tensor, $C_{ijkl}$, in mechanics simulations.
+
+The `RankFourAux` takes as arguments the values of the `index_i`, `index_j`, `index_k`, and `index_l` for the single Rank-4 tensor component to save into an AuxVariable.
+\begin{equation}
+\label{eq:rank4tensor_aux_indices}
+  \begin{aligned}
+        C_{ijkl} \implies & \underbrace{\begin{bmatrix}
+                      C_{11} & C_{12} & C_{13} & C_{14} & C_{15} & C_{16} \\
+                      C_{21} & C_{22} & C_{23} & C_{24} & C_{25} & C_{26} \\
+                      C_{31} & C_{32} & C_{33} & C_{34} & C_{35} & C_{36} \\
+                      C_{41} & C_{42} & C_{43} & C_{44} & C_{45} & C_{46} \\
+                      C_{51} & C_{52} & C_{53} & C_{54} & C_{55} & C_{56} \\
+                      C_{61} & C_{62} & C_{63} & C_{64} & C_{65} & C_{66}
+                      \end{bmatrix}}_{\text{textbook notation}} \\[5.0em]
+         \implies & \underbrace{\begin{bmatrix}
+                      C_{0000} & C_{0011} & C_{0022} & C_{0012} & C_{0020} & C_{0001} \\
+                      C_{1100} & C_{1111} & C_{1122} & C_{1112} & C_{1120} & C_{1101} \\
+                      C_{2200} & C_{2211} & C_{2222} & C_{2212} & C_{2220} & C_{2201} \\
+                      C_{1200} & C_{1211} & C_{1222} & C_{1212} & C_{1220} & C_{1201} \\
+                      C_{2000} & C_{2011} & C_{2022} & C_{2012} & C_{2020} & C_{2001} \\
+                      C_{0100} & C_{0111} & C_{0122} & C_{0112} & C_{0120} & C_{0101}
+                      \end{bmatrix}}_{\text{Rank Four Aux indices}}
```

Comment:
Somehow i wish we could get away from denoting the elasticity tensor with "C", as the letter "C" much more naturally represents the Compliance tensor (the inverse of the elasticity tensor!).

Reply:
@WilkAndy my linear elasticity tensor book tells me that C is for the Stiffness and S is for the Compliance (and blames this squarely on French conventions)...I could add the citation for the use of C to these pages if that would help?

Reply:
Thanks for the info about C and S, @sapitts .   I hope @bwspenc isn't reading this (S=Compliance, C=Stiffness) otherwise he'll have a hernia about naming conventions, heheheh :-)

Reply:
Giving @bwspenc a difficult time about naming conventions has got to be in the fine print of my job description somewhere ;-)

Reply:
This all makes perfect sense!  What are you talking about?

Reply:
It should be consistent with English version of Wikipedia. 

I have never seen S for stiffness and C for compliance although now that I think about it, it makes so much more sense. But tradition is tradition!